lib/pbench/test/unit/agent/task/test_results_push.py

```diff
@@ -23,13 +28,36 @@ class TestResultsPush:
     URL = "http://pbench.example.com/api/v1"
 
     @staticmethod
-    def add_http_mock_response(code: HTTPStatus = HTTPStatus.OK, message: str = ""):
-        responses.add(
-            responses.PUT,
-            f"{TestResultsPush.URL}/upload/{os.path.basename(tarball)}",
-            status=code,
-            json={"message": message},
-        )
+    def add_http_mock_response(
+        status_code: HTTPStatus = None, message: Union[str, Dict, type(None)] = None
+    ):
+        if status_code:
+            if message is None:
+                responses.add(
+                    responses.PUT,
+                    f"{TestResultsPush.URL}/upload/{os.path.basename(tarball)}",
+                    status=status_code,
+                )
+            elif isinstance(message, dict):
+                responses.add(
+                    responses.PUT,
+                    f"{TestResultsPush.URL}/upload/{os.path.basename(tarball)}",
+                    status=status_code,
+                    json=message,
+                )
+            # I can only add a `json' attribute to the response. Trying `text' or `reason' makes
+            # responses.add() barf:
+            #   File "/var/tmp/nick/tox/py39/lib/python3.9/site-packages/responses/__init__.py", line 770, in add
+            #     response = Response(method=method, url=url, body=body, **kwargs)
+            #   File "/var/tmp/nick/tox/py39/lib/python3.9/site-packages/responses/__init__.py", line 563, in __init__
+            #     super().__init__(method, url, **kwargs)
+            # TypeError: __init__() got an unexpected keyword argument 'reason'
+
+        else:
+            responses.add(
+                responses.PUT,
+                f"{TestResultsPush.URL}/upload/{os.path.basename(tarball)}",
+            )
```

Comment:
It seems straightforward enough:
```python
        if status_code:
            if isinstance(message, dict):
                responses.add(
                    responses.PUT,
                    f"{TestResultsPush.URL}/upload/{os.path.basename(tarball)}",
                    status=status_code,
                    json=message,
                )
            elif isinstance(message, (str, BufferedReader, Exception)):
                responses.add(
                    responses.PUT,
                    f"{TestResultsPush.URL}/upload/{os.path.basename(tarball)}",
                    status=status_code,
                    body=message,
                )
            else:
                responses.add(
                    responses.PUT,
                    f"{TestResultsPush.URL}/upload/{os.path.basename(tarball)}",
                    status=status_code,
                )
        else:
            responses.add(
                responses.PUT,
                f"{TestResultsPush.URL}/upload/{os.path.basename(tarball)}",
            )
```
Or, my biases lead me to try a single call with conditional data:
```python
        parms = {}
        if status_code:
            parms["status"] = status_code

        if isinstance(message, dict):
            parms["json"] = message
        elif isinstance(message, (str, BufferedReader, Exception)):
            parms["body"] = message

        responses.add(
            responses.PUT,
            f"{TestResultsPush.URL}/upload/{os.path.basename(tarball)}",
            **parms,
        )
```

Reply:
Yes, that's basically what I did (the first version) but where did the `BufferedReader` and `Exception` cases come from?  I don't understand why they are needed.
 
I'll do one more push using your single call example probably.

Reply:
Those are the types, according to the [documentation](https://github.com/getsentry/responses/blob/master/README.rst#response-parameters), that the `body` parameter will accept.